src/app/clusters/testing/CommandTesting.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/testing/CommandTesting.h>
+
+#include <app/data-model/Encode.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/TLVTags.h>
+#include <lib/support/CodeUtils.h>
+#include <protocols/interaction_model/StatusCode.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SessionManager.h>
+#include <transport/raw/MessageHeader.h>
+
+namespace chip {
+namespace app {
+namespace Testing {
+
+CHIP_ERROR MockCommandHandler::FallibleAddStatus(const ConcreteCommandPath & aRequestCommandPath,
+                                                 const Protocols::InteractionModel::ClusterStatusCode & aStatus,
+                                                 const char * context)
+{
+    StatusRecord record{ aRequestCommandPath, aStatus, context };
+    mStatuses.push_back(std::move(record));
+    return CHIP_NO_ERROR;
+}
+
+void MockCommandHandler::AddStatus(const ConcreteCommandPath & aRequestCommandPath,
+                                   const Protocols::InteractionModel::ClusterStatusCode & aStatus, const char * context)
+{
+    CHIP_ERROR err = FallibleAddStatus(aRequestCommandPath, aStatus, context);
+    VerifyOrDie(err == CHIP_NO_ERROR);
+}
+
+CHIP_ERROR MockCommandHandler::AddResponseData(const ConcreteCommandPath & aRequestCommandPath, CommandId aResponseCommandId,
+                                               const DataModel::EncodableToTLV & aEncodable)
+{
+    chip::System::PacketBufferHandle handle = chip::MessagePacketBuffer::New(chip::kMaxAppMessageLen);
+    VerifyOrReturnError(!handle.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    TLV::TLVWriter baseWriter;
+    baseWriter.Init(handle->Start(), handle->MaxDataLength());
+    DataModel::FabricAwareTLVWriter writer(baseWriter, mFabricIndex);
+    TLV::TLVType ct;
+
+    ReturnErrorOnFailure(static_cast<TLV::TLVWriter &>(writer).StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, ct));
```

Comment:
needing `static_cast` is a code smell ... what is going on here? 

Try to remove the need of a static cast or heavily comment on what is going on. You seem to be relying on `this is a struct with the first member being a TLV writer` ... that seems off. Why not use `writer.mTLVWriter` ?